### PR TITLE
Re-open log files on SIGUSR1 signal

### PR DESCRIPTION
**Pull request by Kami**
With this pull request all the Python service processes now re-open log files when receiving `SIGUSR1` signal.

To test it start the st2api service and run the commands listed bellow.

Screen 1:

```bash
tail -F logs/st2api.log
```

Screen 2:

```bash
python ./st2client/st2client/shell.py runner list - you should see new messages in the log file
rm logs/st2api.log
python ./st2client/st2client/shell.py runner list - you should see no new messages in the log file since the process is still referencing the old file handle
kill -USR1 <st2api pid>
python ./st2client/st2client/shell.py runner list - you should see new messages in the log file because SIGUSR1 causes process to reopen the log files
```

TODO:

- [x] Do the same for all the Python services
- [x] Update log rotate configs
- [x] Remove rotating file handler code

### CODE REVIEW

File: st2common/st2common/logging/misc.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
```
Comment (Kami) [inline st2common/st2common/logging/misc.py]: I'm open to suggestions for a better module name.

**Comment by manasdk** [comment]
Do we need to remove RotatingFileHandler?

**Comment by Kami** [comment]
@manasdk The question is if we need it for the dev environment or not. If we do, we then need two different set of logging configs (dev, prod).

Or we could just leave it there, but update the default values so no rotation happens. This way user needs to explicitly enable it if they want to use that instead of logrotate.

**Comment by manasdk** [comment]
@kami - I guess I don't understand why logrotate is better than RotatingFileHandler.

**Comment by Kami** [comment]
@manasdk Alright, let me try to clarify it again.

> e.g. who calls kill -USR1 <st2api pid>

This would be called by an external program which is responsible for rotating logs. In our case, this is logrotate - we just need to update logrotate.conf to send this signal to the service instead of restarting it.

**Comment by manasdk** [comment]
Oh ok. in postrotate - got it.

So still these are 2 parallel choices - yes/no?

**Comment by Kami** [comment]
@manasdk Yeah.

@jfryman added a log rotate config and we will install this config in the package based deployment which means log rotation will already be taken care of in the packaged based deployments.

**Comment by Kami** [comment]
I've pushed a change which updates log rotate config - f19f0580249683915a31de4741bb0b44e02ee995, 07404817b838e67447249edfec7d79c310ee020b

I also pushed a logging config change which disables log rotation by default. I didn't touch the handler so if user for some reasons wants to use Python and not logrotate to rotate the logs for them, they can do that by updating the logging configs (need to add some docs for the whole log rotation stuff).

**Comment by Kami** [comment]
Added some docs about configuring log rotation - 423d998418ac25c87d9d7310e4915c6283eb72c0.

**Comment by manasdk** [comment]
@Kami - Seems like logrotate wins by default. No use arguing over this any longer.

+1

**Comment by Kami** [comment]
Note / reminder for us - we still need to update packaging manifest files to copy over logrotate config on installation.

**Pull request merged**